v2/middlewares/backoff.go

```diff
@@ -0,0 +1,32 @@
+package middlewares
+
+import (
+	"context"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+
+	backoff_policy "github.com/honestbank/backoff-policy"
+	"github.com/honestbank/kp/v2/internal/middleware"
+)
+
+type backoff struct {
+	p backoff_policy.BackoffPolicy
+}
+
+func (b backoff) Process(ctx context.Context, item *kafka.Message, next func(ctx context.Context, item *kafka.Message) error) error {
+	var err error
+	b.p.Execute(func(marker backoff_policy.Marker) {
+		err = next(ctx, item)
+		if err != nil {
+			marker.MarkFailure()
+			return
+		}
+		marker.MarkSuccess()
+	})
+
+	return err
```

Comment:
Should we probably change the `BackoffPolicy`'s `Execute` function to return an error when it decides to fail ( when reached a maximum attempt or something), then return that error?
This one will return an error even though the retry succeeds.

Before that, it probably makes more sense to return a `nil` here

Reply:
If we allow BackoffPolicy to return certain types, we will end up with use case where sometimes we might need 2 return types, or sometimes it's just the one.

We can't return nil, because that'd mean this middleware would eat the error and make the underlying process never know if something failed.